module/VuFind/src/VuFind/Content/Covers/Orb.php

```diff
@@ -0,0 +1,129 @@
+ * @author   Frédéric Demians <f.demians@tamil.fr>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\Content\Covers;
+
+/**
+ * Orb cover content loader.
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Frédéric Demians <f.demians@tamil.fr>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:content_provider_components
+ */
+class Orb extends \VuFind\Content\AbstractCover
+    implements \VuFindHttp\HttpServiceAwareInterface
+{
+    use \VuFindHttp\HttpServiceAwareTrait;
+
+    /**
+     * Base URL for Orb API
+     *
+     * @var string
+     */
+    protected $url;
+
+    /**
+     * API user for Orb
+     *
+     * @var string
+     */
+    protected $apiUser;
+
+    /**
+     * API key for Orb
+     *
+     * @var string
+     */
+    protected $apiKey;
+
+    /**
+     * Constructor
+     *
+     * @param string $url     Base URL for Orb
+     * @param string $apiUser API key for Orb
+     * @param string $apiKey  API key for Orb
+     */
+    public function __construct($url, $apiUser, $apiKey)
+    {
+        $this->url = $url;
+        $this->apiUser = $apiUser;
+        $this->apiKey = $apiKey;
+        $this->supportsIsbn = $this->cacheAllowed = true;
+    }
+
+    /**
+     * Get an HTTP client
+     *
+     * @param string $url URL for client to use
+     *
+     * @return \Laminas\Http\Client
+     */
+    protected function getHttpClient($url = null)
+    {
+        if (null === $this->httpService) {
+            throw new \Exception('HTTP service missing.');
+        }
+        return $this->httpService->createClient($url);
+    }
+
+    /**
+     * Get image URL for a particular API key and set of IDs (or false if invalid).
+     *
+     * @param string $key  API key
+     * @param string $size Size of image to load (small/medium/large)
+     * @param array  $ids  Associative array of identifiers (keys may include 'isbn'
+     * pointing to an ISBN object and 'issn' pointing to a string)
+     *
+     * @return string|bool
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function getUrl($key, $size, $ids)
+    {
+        if (!isset($ids['isbn'])) {
+            return false;
+        }
+        $ean = $ids['isbn']->get13();
+
+        $url = 'https://' . $this->apiUser . ':' . $this->apiKey . '@' .
+               $this->url . '/products?eans=' . $ean . '&sort=ean_asc';
+        $result = $this->getHttpClient($url)->send();
+        if ($result->isSuccess()) {
+            $data = $result->getBody();
+            $json = json_decode($data, true);
+            foreach ($json['data'] as $title) {
+                if ($title['ean13'] == $ean) {
+                    $size = $size == 'small' ? 'thumbnail' : 'original';
+                    return $title['images']['front'][$size]['src'];
```

Comment:
Is it worth adding an `isset` check here, just in case the API returns something unexpected?

Reply:
I'm just wondering what happens if for some reason the API endpoint returns a data array that lacks an images or front element. Maybe this is impossible. Just wondering if it would be worth having a safety like:

<pre>
if (!isset($title['images']['front'][$size]['src'])) {
    throw new \Exception("Unexpected missing $size image from Orb.");
}
</pre>

...and then if something changes or goes wrong, there will be a record in the error log, rather than a PHP notice and possible unexpected behavior.

Reply:
> I'm just wondering what happens if for some reason the API endpoint returns a data array that lacks an images or front element. Maybe this is impossible. Just wondering if it would be worth having a safety like:

> if (!isset($title['images']['front'][$size]['src'])) {
>     throw new \Exception("Unexpected missing $size image from Orb.");
> }

I will test as suggested, but in order to avoid returning an invalid url.